x-pack/plugin/security/qa/security-basic/src/test/java/org/elasticsearch/xpack/security/apikey/ApiKeyRestIT.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.security.apikey;
+
+import org.elasticsearch.client.Request;
+import org.elasticsearch.client.RequestOptions;
+import org.elasticsearch.client.Response;
+import org.elasticsearch.client.security.support.ApiKey;
+import org.elasticsearch.common.settings.SecureString;
+import org.elasticsearch.common.xcontent.XContentType;
+import org.elasticsearch.test.XContentTestUtils;
+import org.elasticsearch.xpack.core.security.authc.support.UsernamePasswordToken;
+import org.elasticsearch.xpack.security.SecurityInBasicRestTestCase;
+import org.junit.After;
+import org.junit.Before;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.instanceOf;
+import static org.hamcrest.Matchers.notNullValue;
+
+/**
+ * This IT runs in the "security-basic" QA test because it has a working cluster with
+ * API keys enabled, and there's no reason to have a dedicated QA project for API keys.
+ */
+public class ApiKeyRestIT extends SecurityInBasicRestTestCase {
+
+    private static final String SYSTEM_USER = "system_user";
+    private static final SecureString SYSTEM_USER_PASSWORD = new SecureString("sys-pass".toCharArray());
+    private static final String END_USER = "end_user";
+    private static final SecureString END_USER_PASSWORD = new SecureString("user-pass".toCharArray());
+
+    @Before
+    public void createUsers() throws IOException {
+        createUser(SYSTEM_USER, SYSTEM_USER_PASSWORD, List.of("system_role"));
+        createRole("system_role", Set.of("manage_api_key"));
+        createUser(END_USER, END_USER_PASSWORD, List.of("user_role"));
+        createRole("user_role", Set.of("monitor"));
+    }
+
+    @After
+    public void cleanUp() throws IOException {
+        deleteUser("system_user");
+        deleteUser("end_user");
+        deleteRole("system_role");
+        deleteRole("user_role");
+        invalidateApiKeysForUser(END_USER);
+    }
+
+    public void testGrantApiKeyForOtherUser() throws IOException {
```

Comment:
I can't in this test, because basic license doesn't suppor tokens, but I can move the test elsewhere and do that if you'd like it.

Since the transport action test already covers Username+Password and AccessToken (both valid and invalid), I didn't think it was necessary to cover it again in the IT. I just wanted something integrated to test it top-to-bottom.

Testing's a tricky balance, so if it's something you think we need, I'm happy to make it happen (but it will probably means a whole new QA project for rest tests using  security on trial licenses).

Reply:
SGTM to leave it as is

Reply:
> I can't in this test, because basic license doesn't support tokens

But you can start a trial, as it's done in `SecurityWithBasicLicenseIT#testWithTrialLicense` , right ? If this is possible, I would also just slightly prefer a test case using tokens.

Reply:
No, because a cluster can only start a trial once, and the other test does that.
If we want to test this with a non-basic license then we'll need a new QA project.

Reply:
You convinced me that I was just being lazy, so I've added a new QA project, and a new test for authenticating with a token + API key expiration.